Tests/MacTests.swift

```diff
@@ -0,0 +1,95 @@
+        var trustedAppSystemUIServer: SecTrustedApplication?
+        
+        XCTAssertEqual(SecTrustedApplicationCreateFromPath(nil, &trustedAppSelf), errSecSuccess)
+        XCTAssertEqual(SecTrustedApplicationCreateFromPath("/System/Library/CoreServices/SystemUIServer.app", &trustedAppSystemUIServer), errSecSuccess);
+        let trustedList = [trustedAppSelf!, trustedAppSystemUIServer!] as NSArray?
+        
+        // Add an entry to the keychain with an access control list.
+        XCTAssertEqual(SecAccessCreate("Access Control List" as CFString, trustedList, &accessList), errSecSuccess)
+        var accessListQuery = query
+        accessListQuery[kSecAttrAccess as String] = accessList
+        accessListQuery[kSecValueData as String] = vulnValue.data(using: .utf8)
+        XCTAssertEqual(SecItemAdd(accessListQuery as CFDictionary, nil), errSecSuccess)
+        
+        // The potentially vulnerable keychain item should exist in our Valet now.
+        XCTAssertTrue(valet.containsObject(forKey: vulnKey))
+        
+        // Obtain a reference to the vulnerable keychain entry.
+        query[kSecReturnRef as String] = true
+        query[kSecReturnAttributes as String] = true
+        var vulnerableEntryReference: CFTypeRef?
+        XCTAssertEqual(SecItemCopyMatching(query as CFDictionary, &vulnerableEntryReference), errSecSuccess)
+        
+        guard let vulnerableKeychainEntry = vulnerableEntryReference as! NSDictionary? else {
+            XCTFail()
+            return
+        }
+        guard let vulnerableValueRef = vulnerableKeychainEntry[kSecValueRef as String] else {
+            XCTFail()
+            return
+        }
+        
+        let queryWithVulnerableReference = [
+            kSecValueRef as String: vulnerableValueRef
+            ] as CFDictionary
+        // Demonstrate that the item is accessible with the reference.
+        XCTAssertEqual(SecItemCopyMatching(queryWithVulnerableReference, nil), errSecSuccess)
+        
+        // Update the vulnerable value with Valet - we should have deleted the existing item, making the entry no longer vulnerable.
+        let updatedValue = "Safe"
+        XCTAssertTrue(valet.set(string: updatedValue, forKey: vulnKey))
+        
+        // We should no longer be able to access the keychain item via the ref.
+        let queryWithVulnerableReferenceAndAttributes = [
+            kSecValueRef as String: vulnerableValueRef,
+            kSecReturnAttributes as String: true
+            ] as CFDictionary
+        XCTAssertEqual(SecItemCopyMatching(queryWithVulnerableReferenceAndAttributes, nil), errSecItemNotFound)
+        
+        // If you add a breakpoint here then manually inspect the keychain via Keychain.app (search for "MacOSVulnTest"), "xctest" should be the only member of the Access Control list.
+        // This is not be the case upon setting a breakpoint and inspecting before the valet.setString(, forKey:) call above.
```

Comment:
Should this comment have an assertion to go along with it?

Reply:
There's no good way to test the access control list without doing the above steps, unfortunately. If there were then fixing the ACL vuln would have been a lot simpler 😉